web/templates/typeahead_list_item.hbs

```diff
@@ -15,7 +15,21 @@
     {{/if}}
 {{else if is_user_group}}
     <i class="typeahead-image zulip-icon zulip-icon-triple-users no-presence-circle" aria-hidden="true"></i>
+{{else if is_stream_topic}}
+    <div class="typeahead-text-container compose-stream-name">
+        <strong class="typeahead-strong-section">
+            {{~> inline_decorated_stream_name stream=stream_data ~}}
+        </strong>
+        <span role="button" class="conversation-arrow zulip-icon zulip-icon-chevron-right"></span>
+    </div>
+
+    <div class="typeahead-text-container">
+        <strong class="typeahead-strong-section">
+            {{~ topic ~}}
+        </strong>
+    </div>
 {{/if}}
+{{#unless is_stream_topic}}
```

Comment:
I think this would be a bit nicer if you did `if is_stream_topic` and then `else` for the rest of it (instead of `unless`).